common/src/main/java/com/wynntils/wynn/handleditems/annotators/game/DungeonKeyAnnotator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.handleditems.annotators.game;
+
+import com.wynntils.handlers.item.ItemAnnotation;
+import com.wynntils.handlers.item.ItemAnnotator;
+import com.wynntils.mc.utils.ItemUtils;
+import com.wynntils.wynn.handleditems.items.game.DungeonKeyItem;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.ItemStack;
+
+public final class DungeonKeyAnnotator implements ItemAnnotator {
+    private static final Pattern DUNGEON_KEY_PATTERN = Pattern.compile("(?:§.)*(?:Broken )?(?:Corrupted )?(.+) Key");
+
+    public static Matcher dungeonKeyNameMatcher(Component text) {
+        return DUNGEON_KEY_PATTERN.matcher(WynnUtils.normalizeBadString(text.getString()));
+    }
+
+    @Override
+    public ItemAnnotation getAnnotation(ItemStack itemStack) {
+        Matcher keyMatcher = dungeonKeyNameMatcher(itemStack.getHoverName());
```

Comment:
Okay so, we remove the WynnItemMatchers, I assume, and put everything in item stacks. Not sure if there will be a bit of back-fire where it would be useful to have WynnItemMatchers

Reply:
I don't think so. The idea is that the responsibility of determining what there is to know about an item is the responsbility of the ItemModel. If some feature would need to know something more from an item, it should ask the ItemModel to provide it instead.